app/(datasets)/data-catalog/[slug]/page.tsx

```diff
@@ -0,0 +1,72 @@
+import React from 'react';
+import { notFound } from 'next/navigation';
+import { CustomMDX } from 'app/components/mdx';
+import { getDatasets } from 'app/blog/utils/mdx';
+import { baseUrl } from 'app/sitemap';
+import { PageHero } from 'app/lib';
+
+function generateStaticParams() {
+  const posts = getDatasets();
+
+  return posts.map((post) => ({
+    slug: post.slug,
+  }));
+}
+
+export function generateMetadata({ params }) {
+  const post = getDatasets().find((post) => post.slug === params.slug);
+
+  if (!post) {
+    return;
+  }
+
+  const { name, pubDate: publishedTime, description, image } = post.metadata;
+  const ogImage = image
+    ? image
+    : `${baseUrl}/og?title=${encodeURIComponent(name)}`;
+
+  return {
+    name,
+    description,
+    openGraph: {
+      name,
+      description,
+      type: 'article',
+      publishedTime,
+      url: `${baseUrl}/blog/${post.slug}`,
```

Comment:
```suggestion
      url: `${baseUrl}/data-catalog/${post.slug}`,
```

(It will be great if we can find a way to dynamically assign the path for these meta data)

Reply:
also baseUrl is encoded as https://portfolio-blog-starter.vercel.app from this line: https://github.com/developmentseed/next-veda-ui/blob/main/app/sitemap.ts#L3 (We can handle this separately but heads-up)

Reply:
yeah for sure, we should address that in a later iteration or when we run into it again which we will.